### PR TITLE
Initial support for Sandstorm servers added

### PR DESCRIPTION
**Pull request by jparyani**
@RocketChat/core 

This adds basic support for Rocket.Chat on Sandstorm. It is blocking on https://github.com/RocketChat/Rocket.Chat/pull/3437, and to test you will need to use an updated spk (available at https://oasis.sandstorm.io/shared/6ZAPj5yiY1SXs9p8VSLTYQ904IjjBptM9-a5X2MMXS9).

There's still some outstanding issues:
1. Images don't work
2. On loading the app a 2nd time, it crashes (I'm not sure this is due to anything I did since it seems to happen deep in Cordova code).

### CODE REVIEW

File: config.xml [hunk 1]
```diff
@@ -92,6 +92,7 @@
     <preference name="deployment-target" value="8.0" />
     <preference name="AndroidLaunchMode" value="singleTask" />
     <preference name="Orientation" value="all" />
+    <preference name="AppendUserAgent" value="git/" /><!-- 'git' is a whitelisted user agent for Sandstorm -->
```
Comment (kentonv) [inline config.xml]: We don't actually need this anymore, right?

**Comment by kentonv** [comment]
Note that this is WIP -- not ready for merge yet -- due to the second issue. Jason is on vacation next week but it would be nice to get some review in the meantime.

**Comment by jparyani** [comment]
Hi all, sorry for letting this languish. The crash has been fixed (it was due to `fixIndexFile` unexpectedly being called every time the app starts).

Can someone from the Rocket.Chat dev team take a look and let me know how it looks?

**Comment by Sing-Li** [comment]
@RocketChat/core  looks like this is ready to go as a first cut to support Sandstorm clients.   

It no longer crashes.  Please confirm that it has no impact on non-Sandstorm access. 

It would be great if we can merge it in time for the next cut => 0.36.0    Thanks.

**Comment by engelgabriel** [comment]
THIS IS AWESOME!!!!! Thanks @jparyani and @kentonv 

@rodrigok can you please review this?

**Comment by rodrigok** [comment]
The code seems good

**Comment by fadelkon** [comment]
This is not working for me yet with desktop client. Should it? I find it very important to have the desktop client because it's easier to autostart and to keep credentials or sessions. Thank you for collaborating so tightly!

**Versions**
Sandstorm: 0.188
Rocket Chat server: 0.39.0
Rocket Chat debianish desktop client: 1.3.1

**Comment by geekgonecrazy** [comment]
@fadelkon this fix is specifically for the Mobile app.  Please see https://github.com/RocketChat/Rocket.Chat.Electron/issues/97

**Pull request merged**